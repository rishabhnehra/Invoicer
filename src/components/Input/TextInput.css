.TextInput {
    color: #000;
    height: 48px;
    width: 380px;
    box-sizing: border-box;
    padding-left: 18px;
    font-size: 18px;
    border-radius: 5px;
    border: none;
    box-shadow: 0px 0px 6px 3px rgba(0,0,0,0.1);
}

.TextInput:focus { 
    outline: none;
}

.TextInput::placeholder {
    color: rgba(0,0,0,0.3);
    
}