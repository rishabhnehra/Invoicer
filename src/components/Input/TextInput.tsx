import './TextInput.css';

export type TextInputProps = {
    placeholder: string,
    type: string,
    id?: string,
    labelText?: string,
    onClick?: () => void,
    onChange?: () => void
}

export const TextInput: React.FC<TextInputProps> = ({ 
    placeholder,
    id,
    type,
    labelText,
    onClick,
    onChange
}) => {
    return <>
        {labelText && <label htmlFor={id}>{labelText} </label>}
        <input
            className="TextInput"
            id={id}
            placeholder={placeholder}
            type={type}
            onClick={onClick}
            onChange={onChange}
        />
    </>
}