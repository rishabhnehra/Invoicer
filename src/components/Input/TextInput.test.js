import renderer from 'react-test-renderer';
import { TextInput } from './TextInput';

describe("<TextInput />", () => {
    it("snapshot check", () => {
        const tree = renderer.create(<TextInput placeholder="email" type="email" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});