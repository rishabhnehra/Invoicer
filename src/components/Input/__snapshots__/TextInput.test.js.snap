// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> snapshot check 1`] = `
<input
  className="TextInput"
  placeholder="email"
  type="email"
/>
`;
